From 2e83885ca9fa2115c4105b1e0d12a7237ea6a7c8 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Fri, 23 Mar 2018 17:06:55 +0200
Subject: [PATCH 118/706] enetc: udpate coherency configuration per latest BG

The old configuration didn't match the newer spec, it did full cache line
update for BDRs potentially trashing valid data written by CPU.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
(cherry picked from commit 79bd58e099121f7876691d7c1763a54db30f533e)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_hw.h | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_hw.h b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
index 9dc73f64392a..21b8435a3d6b 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_hw.h
+++ b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
@@ -49,8 +49,13 @@
 /* Cache attribute registers for transactions initiated by ENETC */
 #define ENETC_SICAR0	0x08
 #define ENETC_SICAR1	0x0C
-#define ENETC_SICAR_RD_COHERENT	0x2b2b0000 /* rd snoop, no alloc */
-#define ENETC_SICAR_WR_COHERENT	0x00006767 /* wr snoop, no alloc */
+/*
+ * rd snoop, no alloc
+ * wr snoop, no alloc, partial cache line update for BDs and full cache line
+ * update for data
+ */
+#define ENETC_SICAR_RD_COHERENT	0x2b2b0000
+#define ENETC_SICAR_WR_COHERENT	0x00006727
 #define ENETC_SICAR_WR_MSI	0x00000037 /* wr no snoop, no alloc */
 
 #define ENETC_SIPMAR0	0x80
-- 
2.17.1

